Tactics/Simp.lean

```diff
@@ -52,12 +52,15 @@ def LNSymSimpContext
   (exprs : Array Expr := #[])
   -- Simprocs to add to the default set.
   (simprocs : Array Name := #[])
+  -- Whether the default simprocs should be used.
+  (useDefaultSimprocs : Bool := true)
   -- argument to `DiscrTree.mkPath`
   (noIndexAtArgs : Bool := true)
   : MetaM (Simp.Context ×  Array Simp.Simprocs) := do
   let mut ext_simpTheorems := #[]
   let default_simprocs ← Simp.getSimprocs
-  let mut all_simprocs := (#[default_simprocs] : Simp.SimprocsArray)
+  let mut all_simprocs :=
+    if useDefaultSimprocs then #[default_simprocs] else #[]
```

Comment:
Why not something like:
```
let mut all_simprocs := #[]
if useDefaultSimprocs then
  all_simprocs.push <|<- Simp.getSimprocs
```
So we don't needlessly fetch the simprocs if not needed

```

Reply:
We get a weird failure here:

```lean
def fails : MetaM Unit := do
  -- arr : Array arr✝
  let mut arr : Array Bool ← do
      pure #[]
  -- unknown identifier arr
  arr := arr.push false
  return ()

def suceeds : MetaM Unit := do
  let arr : Array Bool ← do
      pure #[]
  let mut arr := arr
  arr := arr.push false
  return ()
```

Reply:
@alexkeizer  I can do:

```
  let all_simprocs : Array Simp.Simprocs ← do
    if useDefaultSimprocs then
      pure #[(← Simp.getSimprocs)]
    else
      pure #[]
  /- For whatever reason, can't have the `all_simprocs` be `mut`. -/
  let mut all_simprocs := all_simprocs
```

if that's better in your view.

Reply:
Addressed,